### ISSUE TITLE
BUG [`@typescript-eslint/no-unused-expressions`]: Rule config is invalid

### ISSUE THREAD

**Issue by dalisoft** [iamturns/eslint-config-airbnb-typescript#198]
## ESLint config
- [airlight-react](https://github.com/dalisoft/airlight/tree/master/packages/eslint-config-react) based on your eslint config

## Info
- OS: macOS 10.15.7
- Node: v15.8.0

## Dependencies
`package.json` (partially shown)
```json
{
    "eslint": "^7.20.0",
    "@typescript-eslint/eslint-plugin": "^4.14.2",
    "eslint-config-airbnb-typescript": "^12.3.1",
    "eslint-config-airlight-react": "^1.1.3",
}
```

## Repro

`.eslintrc`
```json
{
  "extends": "airlight-react"
}
```

## Log

```bash
❯ yarn eslint .
yarn run v1.22.10
$ /Users/dalisoft/Desktop/Workdir/ADV/MyScreen/myscreen-frontend-new/node_modules/.bin/eslint .

Oops! Something went wrong! :(

ESLint: 7.20.0

Error: .eslintrc » eslint-config-airlight-react » eslint-config-airbnb-typescript » /Users/dalisoft/Desktop/Workdir/ADV/MyScreen/myscreen-frontend-new/node_modules/eslint-config-airbnb-typescript/lib/shared.js:
        Configuration for rule "@typescript-eslint/no-unused-expressions" is invalid:
        Value {"allowShortCircuit":false,"allowTernary":false,"allowTaggedTemplates":false,"enforceForJSX":false} should NOT have additional properties.
```

**Comment by duongdev** [comment]
I am getting this issue too.

**Comment by dalisoft** [comment]
@duongdev Yes, downgrading to `eslint@7.19.0` works. Sometimes after restarting IDE starting works (after downgrading)

**Comment by dalisoft** [comment]
It's bug in `eslint@7.20.0` and should be fixed in `eslint@7.31.x` as described other issue (today can't found)
